tools/driver/CMakeLists.txt

```diff
@@ -38,6 +38,11 @@ if(SWIFT_HOST_VARIANT MATCHES "${SWIFT_DARWIN_VARIANTS}")
       LINK_FLAGS " -lobjc -Wl,-force_load,\"${SWIFT_REPL_ARCLITE}\"")
 endif()
 
+if("${CMAKE_SYSTEM_NAME}" STREQUAL "CYGWIN")
+  set_property(TARGET swift APPEND_STRING PROPERTY
+      LINK_FLAGS " -Wl,--allow-multiple-definition")
```

Comment:
Linker generates the error of multiple definition of `guard variable for swift::ImmutablePointerSetFactory<swift::SILInstruction>::EmptyPtrSet` in `libswiftSILOptimizer.a`.
And the `EmptyPtrSet` is defined at `include/swift/Basic/ImmutablePointerSet.h`.

The symbol is defined in the object files compiled from all `lib/SILOptimizer/ARC/*.cpp` (10 files).  I think it is related to 'template instantiation'. But I don't know whether those sources are good or bad.

I added this flag instead of touching sources.